SWIPs/swip-message_to_honey.md

```diff
@@ -13,28 +13,33 @@ Created: 31-07-2019
+In Swarm, nodes send various types of messages; chunk requests, chunk delivery, PSS… to name a few. Every message is priced differently to reflect the different load such messages have on the system. To ensure optimal usage and health of the network it is essential that these prices are quoted relative to the load they induce on the network. If this is not the case certain messages may be too cheap - resulting in unprofitability for service providers, while others may be too expensive - causing users to avoid the service. The Swarm network must be used by real nodes to get insights on how messages should be priced relative to each other. Because of this, there is a need for an update process of the relative prices of messages on a live network. This update process should be atomic: either all nodes upgrade or none, which is currently not facilitated by the hardcoded values in the Swarm source code. Therefore, if in the current implementation prices change, nodes running different releases will account for messages differently. Eventually, this will result in disconnects due to balance differences between nodes. This SWIP proposes an on-chain price oracle to enable an atomic update process of message prices without requiring nodes to update their source-code. 
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for the current Swarm platform and future client implementations.-->
-* The Swarm source-code references the address of a smart-contract which acts as a honey pricing oracle for messages.
-* The price oracle implements the `MsgToHoneyPrice` interface (to-be specified) 
-* The `MsgToHoneyPrice` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
-* Upon receiving/sending of a message, a node checks his local database if his cached price is valid. If not, the node will query the price oracle to get the current price. The node will apply the current price to the balance of his peer. (see diagram below)
+* The Swarm source code references the address of a smart-contract which acts as a honey pricing oracle for messages.
+* The price oracle implements the `MsgToHoney` interface (to-be specified) 
+* The `MsgToHoney` contract will be initially owned by an M/N multi-sig of Swarm developers and stakeholders.
+* Upon receiving/sending of a message, a node checks its local database if its cached price is valid. If not, the node will query the price oracle to get the current price. The node will apply the current price to the balance of its peer (see diagram below).
```

Comment:
local cache, not database (database is implementation specific, we maybe never will really use a database)

Reply:
Ok!